assignments/array-methods.js

```diff
@@ -54,30 +54,85 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
 let fullName = [];
+runners.forEach((element) => {
+    fullName.push(`${element.first_name} ${element.last_name}`);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+allCaps = runners.map((item, index, array) => {
+    return item.first_name.toUpperCase();
+});
+console.log(allCaps);
+
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+
+let largeShirts = runners.filter((item, index, array) => {
+   return item.shirt_size === 'L';
+ });
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((item, index, array) =>{
+  return item += index.donation;
+}, 0);
 console.log(ticketPriceTotal);
 
+
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//find email for each player
+
+let playerEmail = runners.map((item, index, array) => {
+   return item.email;
+ });
+console.log(playerEmail);
 
 // Problem 2
+// Having trouble with this problem. 
+
+// let wrongEmail = runners.filter((item, index, array) => {
+// let emailFormat = [item.email.split()];
+//    for(var i = 0; i < emailFormat.length; i++) {
+//       if(emailFormat === "@") {
+//      }
+//      else {
+//        return item.emailFormat === "@"
+//      }
+//    }
+
+// }
+let wrongEmail = runners.filter((item, index, array) => {
+let emailFormat = [];
+  for(var i = 0; i < emailFormat.length; i++) {
```

Comment:
If I understand what you're trying to do - you're trying to check the list for any emails that aren't formatted correctly? You're on the right track with your if statement. I can't think of a way to check the email formats without using regex - you need to check for more than just the `@` to really verify formats. You might try the `contains` method to check for just that `@` which would get you close enough for now. You can always come back later when you've learned regex.

Reply:
That's exactly what Tramane was telling me, he had a different method as well, which was quite use full. I'll have a try again over the weekend, maybe look into regex by myself. Thanks for the feedback Cole. :)